command/genesis/utils.go

```diff
@@ -288,3 +306,35 @@ func GenerateExtraDataPolyBft(validators []*validator.ValidatorMetadata) ([]byte
 
 	return extra.MarshalRLPTo(nil), nil
 }
+
+// getPricesData fetches the prices for the last 310 days from CoinGecko and unmarshal it
+func getPricesData() (*PricesDataCoinGecko, error) {
+	now := time.Now().UTC()
+	from := now.AddDate(0, 0, -310)
+	apiURL := fmt.Sprintf(
+		`https://api.coingecko.com/api/v3/coins/hydra/market_chart/range?vs_currency=usd&from=%d&to=%d`,
+		from.Unix(),
+		now.Unix(),
+	)
+
+	req, err := common.GenerateThirdPartyJSONRequest(apiURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the key in the header
+	req.Header.Add("x-cg-demo-api-key", "CG-M6fdZBrNeR3njQQtBmkUBhkg")
+
+	body, err := common.FetchPriceData(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var pricesData *PricesDataCoinGecko
+	err = json.Unmarshal(body, &pricesData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	return pricesData, nil
```

Comment:
add a check to ensure we have the data for the past 310 days (the last must be for yesterday)